services/catarse/app/models/project.rb

```diff
@@ -243,6 +244,12 @@ def self.pg_search(term)
     end
   end
 
+  def rewards_have_correct_deliver_at_for_online_project
+    if self.online? && rewards.present? && rewards.select{ |r| r.deliver_at < DateTime.now.beginning_of_day }.present?
```

Comment:
Esse`online?` tu consegue mover para uma condição da validação, ex.: `validate :foo, if: :online?`.  E como o Ton falou, dessa forma que tu fez, tu tá buscando todo mundo no select, iterando como um array e filtrando no ruby, mas como é uma relação has many, tu consegue já fazer tudo no SQL assim:
```suggestion
    if rewards.exists?(deliver_at: ..Time.zone.now)
```